update-versions.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Prepare new artman release.
+
+This script updates SHA hashes of latest commits for both gapic-generator
+and googleapis, and updates version number for subsequent (auto-)release.
+"""
+
+import argparse
+import os.path
+import re
+import requests
+import sys
+from ruamel.yaml import YAML
+
+dockerfile_googleapis_regex = r'^(ENV GOOGLEAPIS_HASH )\S+()'
+dockerfile_gapic_generator_regex = r'^(ENV GAPIC_GENERATOR_HASH )\S+()'
+dockerfile_version_regex = r'^(ENV ARTMAN_VERSION )\S+()'
+setup_py_version_regex = r"^(current_version = ')\S+(')"
+
+github_gapic_generator_project = 'googleapis/gapic-generator'
+github_googleapis_project = 'googleapis/googleapis'
+
+
+def get_latest_commit_sha(project):
+    url = 'https://api.github.com/repos/%s/branches/master' % project
+    # try to get GitHub credentials from config
+    auth = None
+    artman_config = os.path.join(os.environ['HOME'], '.artman', 'config.yaml')
+    try:
+        yaml = YAML()
+        with open(artman_config) as config_yaml:
+            config = yaml.load(config_yaml)
+            github_username = config['github']['username']
+            github_token = config['github']['token']
+            auth = requests.auth.HTTPBasicAuth(github_username, github_token)
+    except:
+        print('Cannot get GitHub token from %s, trying anonymously...' %
+              artman_config)
+
+    sha = None
+    try:
+        response = requests.get(url, auth=auth)
+        repo = response.json()
+        sha = repo['commit']['sha']
+    except:
+        print('Cannot get latest commit SHA. GitHub rate limit?')
+
+    return sha
+
+
+def update_file(filename, regex, version):
+    lines = []
+    with open(filename) as f:
+        for line in f:
+            line = re.sub(regex, r'\g<1>' + version + r'\g<2>', line)
+            lines.append(line)
+    with open(filename, 'w') as f:
+        f.write(''.join(lines))
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Prepare new artman release.')
```

Comment:
It looks like half of this file is argument validation, which feels a little over-engineered to me to just update a couple sha values. I'd really like to treat this as a "script" - just have it do some straightforward actions. I'd propose removing the `--set-X` functionality since we hardly ever need that, and it's trivial to just modify the hashes manually when we need to.